lib/backend/k8s/conversion/conversion_test.go

```diff
@@ -58,6 +59,68 @@ var _ = Describe("Test parsing strings", func() {
 	})
 })
 
+var _ = Describe("Test selector conversion", func() {
+	DescribeTable("selector conversion table",
+		func(inSelector metav1.LabelSelector, selectorType selectorType, expected string) {
+			// First, convert the NetworkPolicy using the k8s conversion logic.
+			c := Converter{}
+
+			converted := c.k8sSelectorToCalico(&inSelector, selectorType)
+
+			// Finally, assert the expected result.
+			Expect(converted).To(Equal(expected))
+		},
+
+		Entry("should handle an empty pod selector", metav1.LabelSelector{}, SelectorPod, "projectcalico.org/orchestrator == 'k8s'"),
+		Entry("should handle an empty namespace selector", metav1.LabelSelector{}, SelectorNamespace, ""),
+		Entry("should handle an OpDoesNotExist namespace selector",
+			metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{Key: "toast", Operator: metav1.LabelSelectorOpDoesNotExist},
+				},
+			},
+			SelectorNamespace,
+			"! has(toast)",
+		),
+		Entry("should handle an OpExists namespace selector",
+			metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{Key: "toast", Operator: metav1.LabelSelectorOpExists},
+				},
+			},
+			SelectorNamespace,
+			"has(toast)",
+		),
+		Entry("should handle an OpIn namespace selector",
+			metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{Key: "toast", Operator: metav1.LabelSelectorOpIn, Values: []string{"butter", "jam"}},
+				},
+			},
+			SelectorNamespace,
+			"toast in { 'butter', 'jam' }",
+		),
+		Entry("should handle an OpNotIn namespace selector",
+			metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{Key: "toast", Operator: metav1.LabelSelectorOpNotIn, Values: []string{"marmite", "milk"}},
+				},
+			},
+			SelectorNamespace,
+			"toast not in { 'marmite', 'milk' }",
```

Comment:
nit:  I think marmite is just as much in toast than butter and jam.  More so IMHO.